### ISSUE TITLE
remove need for sudo

### ISSUE THREAD

**Issue by nlf** [nlf/dlite#172]
this is being done by splitting all of the features that require sudo into their own daemon process with an RPC channel and rewriting the `dlite` command to make calls to the privileged daemon.

this change eliminates the need for modifying `/etc/sudoers` as well.

roadblocks:
- [ ] ~~libxhyve appears to not free the pty when shutting down the vm. this is minor, but does mean that every time the vm is started it will get a new pty.~~ changed tactics, no longer an issue

**Comment by max-mapper** [comment]
I read https://github.com/mist64/xhyve/issues/60 but it's unclear if code singing xhyve (in your case libxhyve) can make it possible to use xhyve with networking without sudo, is that possible?

**Comment by nlf** [comment]
so, from what i've gathered from an apple employee, the required entitlement to avoid requiring sudo is only granted in special cases right now. it would require turning this into an app store compliant app, registering an apple dev account (which i can probably get already) and then getting that entitlement approved.

tl;dr - not feasibly, no

**Comment by nlf** [comment]
also hi max :D
